src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -32,7 +30,7 @@
 WindowsErrorReporting.SetErrorModeOnWindows();
 
 var parser = CreateCommandLineParser();
-return await parser.InvokeAsync(args);
+return await parser.Parse(args).InvokeAsync(CancellationToken.None);
```

Comment:
What cancellation token should be going in here? Was the old behavior listening to Ctrl+C to raise the token or not?

Reply:
The docs show `InvokeAsync` with no cancellationtoken.  Might be one of our analyzers telling it to pass a cancellation token?
https://learn.microsoft.com/en-us/dotnet/standard/commandline/handle-termination

Reply:
Whatever token we pass here, S.CL is going to create a source linked to it:

https://github.com/dotnet/command-line-api/blob/8f2ada7fbaab300dc3fcaa9141d1d69340e3b4cb/src/System.CommandLine/Invocation/InvocationPipeline.cs#L19

and create it's own token out of it, then request the cancellation when its Ctrl+C is pressed

https://github.com/dotnet/command-line-api/blob/8f2ada7fbaab300dc3fcaa9141d1d69340e3b4cb/src/System.CommandLine/Invocation/ProcessTerminationHandler.cs#L85